src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/EmbeddedLanguages/VirtualChars/VirtualCharSequence.cs

```diff
@@ -80,6 +82,9 @@ public VirtualCharSequence GetSubSequence(TextSpan span)
         public VirtualChar First() => this[0];
         public VirtualChar Last() => this[^1];
 
+        public IEnumerable<VirtualChar> AsEnumerable()
+            => new Enumerable(this);
```

Comment:
i will doc.  the general idea here is that VirtualCharSequence is used in hotspots where i don't want accidental use of Linq.  Doing this makes it intentional that you're moving into the allocation/interface-heavy space of linq.  Since htis is only used for the *fixer* part of this PR, that's fine.

Reply:
actually, i just removed linq from this entirely :)